diff --git a/make/devkit/Makefile b/make/devkit/Makefile
index c85a7c21d29..8f69d23c325 100644
--- a/make/devkit/Makefile
+++ b/make/devkit/Makefile
@@ -58,7 +58,7 @@
 COMMA := ,
 
 os := $(shell uname -o)
-cpu := $(shell uname -p)
+cpu := $(shell uname -m)
 
 # Figure out what platform this is building on.
 me := $(cpu)-$(if $(findstring Linux,$(os)),linux-gnu)
diff --git a/make/devkit/Tools.gmk b/make/devkit/Tools.gmk
index 187320ca26e..9024b14a7fe 100644
--- a/make/devkit/Tools.gmk
+++ b/make/devkit/Tools.gmk
@@ -88,7 +88,7 @@ else ifeq ($(BASE_OS), Fedora)
     endif
   endif
   LINUX_VERSION := Fedora_$(BASE_OS_VERSION)
-else
+else ifneq ($(BASE_OS), RHEL)
   $(error Unknown base OS $(BASE_OS))
 endif
 
